From b78f15d194b6db5318e7a9f34117b0886f1bb49d Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 5 Jun 2026 15:17:26 +1100
Subject: [PATCH 2/2] fix pytorch version comparison

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 99f1a80..ddaf0d2 100644
--- a/setup.py
+++ b/setup.py
@@ -268,7 +268,7 @@ def setup_kwargs() -> tp.Dict[str, tp.Any]:
 
     # CUB needed to build the cuAEV, download it if not found bundled with Torch
     include_paths_kwargs: tp.Dict[str, tp.Any]
-    if float(".".join(torch.__version__.split(".")[:2])) >= 2.7:
+    if tuple(map(int, torch.__version__.split(".")[:2])) >= (2, 7):
         include_paths_kwargs = {"device_type": "cuda"}
     else:
         include_paths_kwargs = {"cuda": True}
