diff --git a/setup.py b/setup.py
index 108cd73..043389f 100644
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,7 @@
 import os
 from packaging import version
 from setuptools import setup, find_packages
-from distutils import log
+import warnings
 import sys
 
 BUILD_CUAEV_ALL_SM = '--cuaev-all-sms' in sys.argv
@@ -18,7 +18,7 @@ if ONLY_BUILD_SM80:
     sys.argv.remove('--only-sm80')
 
 if not BUILD_CUAEV_ALL_SM and not FAST_BUILD_CUAEV:
-    log.warn("Will not install cuaev")  # type: ignore
+    warnings.warn("Will not install cuaev")  # type: ignore
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
diff --git a/torchani/__init__.py b/torchani/__init__.py
index a94853d..b19a83b 100644
--- a/torchani/__init__.py
+++ b/torchani/__init__.py
@@ -36,11 +36,15 @@ from . import utils
 from . import neurochem
 from . import models
 from . import units
-from pkg_resources import get_distribution, DistributionNotFound
+from importlib.metadata import version
 import warnings
 
+
+class DistributionNotFound(Exception):
+    pass
+
 try:
-    __version__ = get_distribution(__name__).version
+    __version__ = version(__name__)
 except DistributionNotFound:
     # package is not installed
     pass
diff --git a/torchani/neurochem/parse_resources.py b/torchani/neurochem/parse_resources.py
index 8017f86..dba0a59 100644
--- a/torchani/neurochem/parse_resources.py
+++ b/torchani/neurochem/parse_resources.py
@@ -2,7 +2,7 @@ import os
 import io
 import requests
 import zipfile
-from distutils import dir_util
+import shutil
 from pathlib import Path
 
 
@@ -47,8 +47,8 @@ def parse_neurochem_resources(info_file_path):
                 resource_zip.extractall(local_dir)
                 resource_path = local_dir
             source = os.path.join(resource_path, extracted_name, "resources")
-            dir_util.copy_tree(source, resource_path)
-            dir_util.remove_tree(os.path.join(resource_path, extracted_name))
+            shutil.copytree(source, resource_path, dirs_exist_ok=True)
+            shutil.rmtree(os.path.join(resource_path, extracted_name))
 
         else:
             raise ValueError('File {0} could not be found either in {1} or {2}\n'
